warehouse/models/intermediate/gtfs_quality/guidelines_checks/int_gtfs_quality__scheduled_trips_in_tu_feed.sql

```diff
@@ -16,55 +21,84 @@ dim_provider_gtfs_data AS (
+    FROM fct_daily_scheduled_trips AS scheduled_trips
     LEFT JOIN fct_observed_trips AS observed_trips
-      ON idx.date = observed_trips.dt
-     AND quartet.associated_schedule_gtfs_dataset_key = observed_trips.schedule_to_use_for_rt_validation_gtfs_dataset_key
-     AND scheduled_trips.trip_id = observed_trips.trip_id
+      -- should this be activity date or service date?
+      ON scheduled_trips.service_date = observed_trips.dt
+      AND scheduled_trips.gtfs_dataset_key = observed_trips.schedule_to_use_for_rt_validation_gtfs_dataset_key
+      AND scheduled_trips.trip_id = observed_trips.trip_id
+    GROUP BY 1, 2, 3, 4, 5
+),
 
-     GROUP BY 1, 2
+map_trips_to_services AS (
+    SELECT
+        idx.date,
+        idx.service_key,
+        idx.gtfs_dataset_key,
+        LOGICAL_OR(quartet.schedule_gtfs_dataset_key IS NOT NULL) AS quartet_has_schedule,
+        LOGICAL_OR(quartet.trip_updates_gtfs_dataset_key IS NOT NULL) AS quartet_has_tu,
+        SUM(compare_trips.observed_trips) AS observed_trips,
+        SUM(compare_trips.scheduled_trips) AS scheduled_trips,
+    FROM guideline_index AS idx
+    LEFT JOIN dim_provider_gtfs_data AS quartet
+        ON CAST(idx.date AS TIMESTAMP) BETWEEN quartet._valid_from AND quartet._valid_to
+        AND quartet.service_key = idx.service_key
+        AND (idx.gtfs_dataset_key = quartet.schedule_gtfs_dataset_key
+            OR idx.gtfs_dataset_key = quartet.trip_updates_gtfs_dataset_key
+            OR idx.gtfs_dataset_key = quartet.vehicle_positions_gtfs_dataset_key
+            OR idx.gtfs_dataset_key = quartet.service_alerts_gtfs_dataset_key)
+    LEFT JOIN compare_trips
+        ON idx.date = compare_trips.date
+        AND quartet.schedule_gtfs_dataset_key = compare_trips.schedule_gtfs_dataset_key
+        AND quartet.trip_updates_gtfs_dataset_key = compare_trips.tu_gtfs_dataset_key
+    WHERE idx.service_key IS NOT NULL
+    GROUP BY 1, 2, 3
 ),
 
 int_gtfs_quality__scheduled_trips_in_tu_feed AS (
     SELECT
-        service_key,
-        date,
-        {{ scheduled_trips_in_tu_feed() }} AS check,
-        {{ fixed_route_completeness() }} AS feature,
-        scheduled_trips,
-        observed_trips,
+        idx.* EXCEPT(status),
+        map_trips_to_services.quartet_has_tu,
+        map_trips_to_services.quartet_has_schedule,
+        map_trips_to_services.observed_trips,
+        map_trips_to_services.scheduled_trips,
         CASE
-            WHEN
-                NOT(has_tu_feed)
-                OR scheduled_trips = 0
-                OR scheduled_trips IS null
-                THEN {{ guidelines_na_check_status() }}
-            WHEN scheduled_trips = observed_trips THEN {{ guidelines_pass_status() }}
-            ELSE {{ guidelines_fail_status() }}
+            WHEN has_service AND has_schedule_feed
+                THEN
+                    CASE
+                        WHEN scheduled_trips = observed_trips THEN {{ guidelines_pass_status() }}
```

Comment:
so this just compares counts?

Reply:
We discussed offline it compares counts after it's been joined on `trip_id` so it is checking for trip-level matches